openedx/core/djangoapps/xblock/rest_api/views.py

```diff
@@ -226,7 +226,9 @@ def post(self, request, usage_key_str):
         old_metadata = block.get_explicitly_set_fields_by_scope(Scope.settings)
         old_content = block.get_explicitly_set_fields_by_scope(Scope.content)
 
-        block.data = data
+        # only update data if it was passed
+        if data is not None:
```

Comment:
@rpenido Can you add a test for this change?

Reply:
@ChrisChV I didn't find the xblock api test suit to update with this, so I'm afraid we don't have one.
Do you think we should create one from scratch as part of this task? Should we try to get some budget for that?

Reply:
@rpenido I thought it would be easier :). Being a small change in the rest_api, spending more time than the budget here is unnecessary.